### ISSUE TITLE
Generic constraints are dropped from decompiled signatures, producing invalid C# reported as Full

### ISSUE THREAD

**Issue by richlander** [richlander/dotnet-inspect#3664]
## Summary

The decompiler never emits `where` clauses. A generic method whose body depends on a type-parameter constraint therefore decompiles to C# that does not compile, and the result is reported as `DecompilationFidelity.Full`.

This is the same failure *class* as #3631 — invalid output presented as faithful — but a completely different cause, and it is **pre-existing and unrelated to local functions**. Found while probing the constraint axis of #3643.

## Reproduction

```csharp
using System;
public class Plain {
    public static int Compare<T>(T a, T b) where T : IComparable<T> => a.CompareTo(b);
}
```

Compile, then:

```bash
dotnet run --project src/dotnet-inspect -c Release -- member --library p.dll Plain Compare -S "Decompiled Source"
```

Output:

```csharp
public static int Compare<T>(T a, T b) => a.CompareTo(b);
```

The `where T : IComparable<T>` is gone, so `T` has no `CompareTo`. Compiling that output with `csc`:

```text
out.cs(4,49): error CS7036: There is no argument given that corresponds to the required parameter
'comparisonType' of 'MemoryExtensions.CompareTo(ReadOnlySpan<char>, ReadOnlySpan<char>, StringComparison)'
```

(It binds to an unrelated extension method and then fails, which is if anything worse than a clean "no such member".)

Reported fidelity, measured directly through `IrImporter.Import` + `CSharpPrinter.PrintRaised`:

```text
Plain.Compare: fidelity=Full
```

## Scope

- No local function is involved. It reproduces on a plain generic method.
- It reproduces on clean `origin/main`.
- Constraints are dropped unconditionally — this is not a decode failure, the clauses are simply never printed.

## Why it matters

`Full` is the tool's assertion that the decompilation is complete and faithful. A constraint-bearing generic method is common in real framework code, so this is a broad source of silently invalid output. It also interacts with local-function raising: a raised local function's declaration inherits the host's type parameters, and its own constraints are dropped by construction (a `LocalFunctionStatement` has no type-parameter list). That direction is sound — a call site `Own<T>(…)` is only legal if the host's constraints imply the local's — but it is only sound *given* the host's constraints are printed, which today they are not.

## Suggested direction

Emit `where` clauses for method and type generic parameters from the metadata `GenericParameterConstraint` table (class/struct/new() flags plus constraint types). If a constraint cannot be decoded or spelled, that is a fidelity downgrade, not something to omit silently.
